### PR TITLE
Basic Trusts System

### PR DESCRIPTION
**Pull request by JoshuaTyree**
Trust systems patch. Adds basic party support and more.

What can it do?
- Automatically summon trusts when you start the bot and in between fights
- Check for resources, re-summon if they not have enough to continue
- Determine which party members are trusts
- Determine if you have party members or not

How to make it better
- Split the dismiss trust/summon trust into two different states.  I.E. (ReleaseTrustsState, SummonTrustsState).  This would allow for cleaner state separation to prevent possible time-based issues.
- Classify all trust spells into roles.  (Healer, Tank, Buffer, Nuker, etc) Then allow the player to specify multiple of the same role with a priority system.  If the desired trust is not off cooldown, summon the next in line for a shorter downtime.
- ??? open to suggestions.

### CODE REVIEW

File: EasyFarm/States/SummonTrustsState.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            }
+
+            var command = string.Format("/refa {0}", comp);
+            fface.Windower.SendString(command);
+        }
+
+        public override bool Check() {
+            if (new RestState(fface).Check()) return false;
+            if (
+                fface.Player.Status.Equals(Status.CatchMonster) ||
```
Comment (Mykezero) [inline EasyFarm/States/SummonTrustsState.cs]: We can use a blacklist here `if(!fface.Player.Status.Equals(Status.Standing))` to simplify the check. 

The `Helpers.ToStatus` method in the MemoryAPI project has a list of which statuses are currently in use. The other statuses are returned as `Status.Unknown`. 

[EasyFarm/MemoryAPI/Helpers.cs](https://github.com/EasyFarm/EasyFarm/blob/4962b76cccb168c46ddad2e757417aebb8273a9f/MemoryAPI/Helpers.cs#L24)

**Comment by Mykezero** [comment]
Sorry for the late response; I really want to give this pull request the time it deserves!
I've installed the retail game client, and I'm hoping to test it tonight.

File: EasyFarm/States/SummonTrustsState.cs (hunk 1 above)
Comment (JoshuaTyree) [inline EasyFarm/States/SummonTrustsState.cs]: I see, I'll implement that change.

**Comment by JoshuaTyree** [comment]
I've recently had to pleasure of leveling my THF. I used the PR all the way to 99. When previously I would die after a while due to tanks/healers running out of MP/HP, this did not happen once.  

When testing, test the lower levels.  They'll give you the best picture of what this does.

**Comment by JoshuaTyree** [comment]
I added some more measures to before transitioning to a combat state before ensuring all trusts are summoned/taken care of. This fixes an issue where it would dismiss a trust then before resummoning, it would whack another enemy.

**Comment by Mykezero** [comment]
I'm actually starting to level my thf now xD. I need to get to level 5 before I can test the pull request T_T

**Comment by Mykezero** [comment]
@JoshuaTyree  I had a chance to test this last night and it works great! 

The role-base summoning idea would be a great improvement, but we would probably have to classify them ourselves or allow the user to choose the roles. I'll create the issue for that since it would be nice to have. 

I'll merge the pull request in now and do a little bit more testing as well.

**Comment by Mykezero** [comment]
Thank you for your contribution 👍 It's now available as [EasyFarm-1.0.110](https://github.com/EasyFarm/EasyFarm/releases/tag/EasyFarm-1.0.110)

I really appreciate the time you took in making this happen.  Great Job! 😉

**Pull request merged**